frappe/core/doctype/deleted_document/deleted_document.py

```diff
@@ -27,4 +36,47 @@ def restore(name):
 	deleted.restored = 1
 	deleted.db_update()
 
-	frappe.msgprint(_('Document Restored'))
+	if alert:
+		frappe.msgprint(_('Document Restored'))
+
+
+@frappe.whitelist()
+def bulk_restore(docnames):
```

Comment:
This API method should return a JSON response with a list of succeeded and failed documents. The client-side should handle showing the dialog.